### PR TITLE
PLANET-7445: Migrate Split Two Columns blocks to Columns blocks

### PR DESCRIPTION
**Pull request by mardelnet**
Ref: https://jira.greenpeace.org/browse/PLANET-7445

<hr>

**DESCRIPTION:**
This PR migrates the Split Two Columns blocks to Columns blocks.

<hr>

**RELATED PR:**
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1261

<hr>

**TESTING INSTRUCTIONS:**
1. Run this PR in your local environment.
2. Add a Split Two Columns block to a post or page. it won't work on the backend because of this PR's changes but it's good enough to test the migration script. You can also directly copy/paste this in the code editor:

```
<!-- wp:planet4-blocks/split-two-columns {"title":"This is issue name","issue_description":"And this is the description","issue_link_text":"A test link","issue_link_path":"https://www.greenpeace.org/","select_tag":6,"tag_name":"Climate","tag_description":"Our climate is sick but we have the power to heal it and ourselves, and create a greener more equal world.\r\rExtreme weather events, sea level rise and droughts are all symptoms of a global climate on the brink of breakdown.\r\rWe are at a crossroads. If we act now, as a global community, to keep temperature rise below catastrophic levels, we have a chance to re-stabilise our delicate climate.\r\rWe ca","tag_link":"http://www.planet4.test/tag/climate/","button_text":"Get involved","button_link":"http://www.planet4.test/tag/climate/","tag_image_id":344,"edited":{"title":true,"issue_description":true,"issue_link_text":true,"tag_description":false,"button_text":false,"issue_image_id":false,"tag_image_id":false}} -->
<div class="wp-block-planet4-blocks-split-two-columns"><div data-hydrate="planet4-blocks/split-two-columns" data-attributes="{&quot;version&quot;:2,&quot;select_issue&quot;:0,&quot;title&quot;:&quot;This is issue name&quot;,&quot;issue_description&quot;:&quot;And this is the description&quot;,&quot;issue_link_text&quot;:&quot;A test link&quot;,&quot;issue_link_path&quot;:&quot;https://www.greenpeace.org/&quot;,&quot;issue_image_id&quot;:0,&quot;issue_image_src&quot;:&quot;&quot;,&quot;issue_image_srcset&quot;:&quot;&quot;,&quot;issue_image_title&quot;:&quot;&quot;,&quot;focus_issue_image&quot;:&quot;50% 50%&quot;,&quot;select_tag&quot;:6,&quot;tag_name&quot;:&quot;Climate&quot;,&quot;tag_description&quot;:&quot;Our climate is sick but we have the power to heal it and ourselves, and create a greener more equal world.\r\rExtreme weather events, sea level rise and droughts are all symptoms of a global climate on the brink of breakdown.\r\rWe are at a crossroads. If we act now, as a global community, to keep temperature rise below catastrophic levels, we have a chance to re-stabilise our delicate climate.\r\rWe ca&quot;,&quot;tag_link&quot;:&quot;http://www.planet4.test/tag/climate/&quot;,&quot;button_text&quot;:&quot;Get involved&quot;,&quot;button_link&quot;:&quot;http://www.planet4.test/tag/climate/&quot;,&quot;tag_image_id&quot;:344,&quot;tag_image_src&quot;:&quot;&quot;,&quot;tag_image_srcset&quot;:&quot;&quot;,&quot;tag_image_title&quot;:&quot;&quot;,&quot;focus_tag_image&quot;:&quot;50% 50%&quot;,&quot;edited&quot;:{&quot;title&quot;:true,&quot;issue_description&quot;:true,&quot;issue_link_text&quot;:true,&quot;tag_description&quot;:false,&quot;button_text&quot;:false,&quot;issue_image_id&quot;:false,&quot;tag_image_id&quot;:false}}"><section class="alignfull split-two-column "><div class="split-two-column-item item--left"><div class="split-two-column-item-content"><h2 class="split-two-column-item-title"><a href="https://www.greenpeace.org/" data-ga-category="Split Two Columns" data-ga-action="Category Title" data-ga-label="n/a">This is issue name</a></h2><p class="split-two-column-item-subtitle">And this is the description</p><a class="split-two-column-item-link" href="https://www.greenpeace.org/" data-ga-category="Split Two Columns" data-ga-action="Text Link" data-ga-label="n/a">A test link</a></div></div><div class="split-two-column-item item--right"><div class="split-two-column-item-content"><a class="split-two-column-item-tag" href="http://www.planet4.test/tag/climate/" data-ga-category="Split Two Columns" data-ga-action="Tag Title" data-ga-label="n/a"><span aria-label="hashtag">#</span>Climate</a><p class="split-two-column-item-subtitle">Our climate is sick but we have the power to heal it and ourselves, and create a greener more equal world.

Extreme weather events, sea level rise and droughts are all symptoms of a global climate on the brink of breakdown.

We are at a crossroads. If we act now, as a global community, to keep temperature rise below catastrophic levels, we have a chance to re-stabilise our delicate climate.

We ca</p><a class="btn btn-primary split-two-column-item-button" href="http://www.planet4.test/tag/climate/" data-ga-category="Split Two Columns" data-ga-action="Call to Action" data-ga-label="n/a">Get involved</a></div></div></section></div></div>
<!-- /wp:planet4-blocks/split-two-columns -->

<!-- wp:planet4-blocks/split-two-columns {"button_text":"Get involved"} -->
<div class="wp-block-planet4-blocks-split-two-columns"><div data-hydrate="planet4-blocks/split-two-columns" data-attributes="{&quot;version&quot;:2,&quot;select_issue&quot;:0,&quot;title&quot;:&quot;&quot;,&quot;issue_description&quot;:&quot;&quot;,&quot;issue_link_text&quot;:&quot;&quot;,&quot;issue_link_path&quot;:&quot;&quot;,&quot;issue_image_id&quot;:0,&quot;issue_image_src&quot;:&quot;&quot;,&quot;issue_image_srcset&quot;:&quot;&quot;,&quot;issue_image_title&quot;:&quot;&quot;,&quot;focus_issue_image&quot;:&quot;50% 50%&quot;,&quot;select_tag&quot;:0,&quot;tag_name&quot;:&quot;&quot;,&quot;tag_description&quot;:&quot;&quot;,&quot;tag_link&quot;:&quot;&quot;,&quot;button_text&quot;:&quot;Get involved&quot;,&quot;button_link&quot;:&quot;&quot;,&quot;tag_image_id&quot;:0,&quot;tag_image_src&quot;:&quot;&quot;,&quot;tag_image_srcset&quot;:&quot;&quot;,&quot;tag_image_title&quot;:&quot;&quot;,&quot;focus_tag_image&quot;:&quot;50% 50%&quot;,&quot;edited&quot;:{&quot;title&quot;:false,&quot;issue_description&quot;:false,&quot;issue_link_text&quot;:false,&quot;tag_description&quot;:false,&quot;button_text&quot;:false,&quot;issue_image_id&quot;:false,&quot;tag_image_id&quot;:false}}"><section class="alignfull split-two-column "><div class="split-two-column-item item--left"><div class="split-two-column-item-content"></div></div><div class="split-two-column-item item--right"><div class="split-two-column-item-content"></div></div></section></div></div>
<!-- /wp:planet4-blocks/split-two-columns -->
```

4. Run the migrator command: `npx wp-env run cli wp p4-run-activator`
5. Check whether the migration worked as expected.
6. To reset the environment database back to the latest default content and repeat the tests, run: `npm run db:reset`.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [titan](https://www-dev.greenpeace.org/test-titan) | [admin](https://www-dev.greenpeace.org/test-titan/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-titan/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-titan) | [composer-local.json](https://github.com/greenpeace/planet4-test-titan/blob/main/composer-local.json)

:watch: 2024.11.05 13:38:27

File: src/Migrations/M032MigrateSplit2ColumnBlock.php
Comment (mleray) [inline src/Migrations/M032MigrateSplit2ColumnBlock.php:1]: We need to keep in mind that you're using the same migration number (032) in your Content Covers PR

**Comment by mardelnet** [comment]
> Looks good. Also tested it locally and the migration outcome looks exactly as it should 🚀

@comzeradd, I dismissed your approval since the PR was refactored according to [this discussion](https://github.com/greenpeace/planet4-master-theme/pull/2436#issuecomment-2464251688).
@mleray now you can rebase #2436 with this PR and continue working. As you can see in [this file](https://github.com/greenpeace/planet4-master-theme/blob/43548456525cd19aa0567182719d28fc0a320751/src/Migrations/Utils/Functions.php), I added a bunch of new functions to facilitate the block creation and avoid duplicated code.

File: src/BlockSettings.php [hunk 1]
```diff
@@ -219,7 +219,6 @@ public function set_allowed_block_types(array|bool $allowed_block_types, object
             self::PAGE_BLOCK_TYPES,
             ! Features::is_active(self::BETA_BLOCKS_FEATURE) ? [] : self::BETA_PAGE_BLOCK_TYPES,
             ! $enform_active ? [] : [ self::P4_BLOCKS_PREFIX . '/enform' ],
-            (bool) planet4_get_option('new_ia') ? [] : [ self::P4_BLOCKS_PREFIX . '/split-two-columns' ],
```
Comment (mleray) [inline src/BlockSettings.php:222]: We need to create a follow-up ticket to remove the rest of the relevant code

Comment (mleray) [reply src/BlockSettings.php:222]: Ah, nevermind, it seems it's been done already since it wasn't moved to the theme 👍

**Comment by mleray** [comment]
Hey @mardelnet, when using the code you provided in the editor and running the migration script I get the following error:
```
[11-Nov-2024 09:35:00 UTC] PHP Fatal error:  Uncaught TypeError: P4\MasterTheme\Migrations\Utils\Functions::create_media_text_paragraph(): Argument #4 ($img_id) must be of type int, string given, called in /var/www/html/wp-content/themes/planet4-master-theme/src/Migrations/M032MigrateSplit2ColumnBlock.php on line 161 and defined in /var/www/html/wp-content/themes/planet4-master-theme/src/Migrations/Utils/Functions.php:269
```

**Comment by mardelnet** [comment]
> Hey @mardelnet, when using the code you provided in the editor and running the migration script I get the following error:

Can you re-check it, @mleray? I introduced some new changes.

**Review by mleray — APPROVED** [review]
Tested on local, works as expected 👍

**Comment by mardelnet** [comment]
> Tested on local, works as expected 👍

Hey @mleray 
Would you mind re-checking the PR once again?
As you can see [here](https://github.com/greenpeace/planet4-master-theme/pull/2433/commits/167c3626b7930fa1eb864bb066166c9e45d0c7da), I had to make even more code reusable to avoid conflicts with SonarCloud.

**Comment by mleray** [comment]
Tested again on local with the latest changes and all is good 👏

**Pull request merged**